qiskit/visualization/__init__.py

```diff
@@ -17,20 +17,160 @@
+development environment:
+
+.. code-block:: bash
+
+   pip install 'qiskit[visualization]'
+
 Counts and State Visualizations
 ===============================
 
+This section contains functions that visualize measurement outcome counts and quantum states.
+
+APIs
+----
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
 .. autosummary::
    :toctree: ../stubs/
 
    plot_histogram
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+   :toctree: ../stubs/
+
    plot_bloch_vector
    plot_bloch_multivector
    plot_state_city
    plot_state_hinton
    plot_state_paulivec
    plot_state_qsphere
 
+Example Usage
+-------------
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_histogram` to visualize measurement outcome counts:
+
+.. jupyter-execute::
+
+   from qiskit.tools.visualization import plot_histogram
+
+   counts = {"00": 501, "11": 499}
+   plot_histogram(counts)
+
+The data can be a dictionary with bit string as key and counts as value, or more commonly a
+:class:`~qiskit.result.Counts` object obtained from :meth:`~qiskit.result.Result.get_counts`.
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_state_city` to visualize a quantum state:
+
+.. jupyter-execute::
+
+   from qiskit.tools.visualization import plot_state_city
+
+   state = [[0+1.0j, 0-1.0j],
+            [1-0.5j, 1+0.5j]]
+   plot_state_city(state)
+
+The state can be array-like list of lists, ``numpy.array``, or more commonly
+:class:`~qiskit.quantum_info.Statevector` or :class:`~qiskit.quantum_info.DensityMatrix` objects
+obtained from a :class:`~qiskit.circuit.QuantumCircuit`:
+
+.. jupyter-execute::
+
+   from qiskit import QuantumCircuit
+   from qiskit.quantum_info import Statevector, DensityMatrix
+
+   qc = QuantumCircuit(2)
+   qc.h(0)
+   qc.cx(0,1)
+
+   # plot using a Statevector
+   state = Statevector(qc)
+   plot_state_city(state)
+
+.. jupyter-execute::
+
+   # plot using a DensityMatrix
+   state = DensityMatrix(qc)
+   plot_state_city(state)
+
+You can find code examples for each visualization functions on the individual function `API page
+<#apis>`_.
+
+Common Keyword Arguments
+------------------------
+
+The figures created by counts and state visualizations functions are mostly generated by
+`Matplotlib <https://matplotlib.org/>`_. As a result there are keyargs inherited from
+``Matplotlib`` that are common to most functions, including:
+
+* ``title`` (``str``): a text string to use for the plot title.
+* ``legend`` (``list``): a list of strings to use for labels of the data.
+* ``figsize`` (``tuple``): figure size in inches .
+* ``color`` (``list``): a list of strings for plotting.
+* ``ax`` (`Matplotlib.axes.Axes <https://matplotlib.org/stable/api/axes_api.html>`_): An optional
+  ``Axes`` object to be used for the visualization output. If none is specified a new
+  `Matplotlib.figure.Figure <https://matplotlib.org/stable/api/figure_api.html>`_ will be created
```

Comment:
It's a very small thing but shouldn't `matplotlib` be written in lowercase here? We never write `Qiskit.QuantumCircuit` as far as I know.

Reply:
Very good point! Corrected them.